drivers/gpio/wch_gpio_ch32v00x.c

```diff
@@ -0,0 +1,144 @@
+/*
```

Comment:
I was checking the specs for ch32v307, but I think this implementation can be used for both 003 and 307. Maybe you should reconsider the file name.

Reply:
More changes (and drivers) and support for other chips are coming in future PRs. We can refactor later as required.